cmd/bootstrap/run/epochs.go

```diff
@@ -0,0 +1,200 @@
+package run
+
+import (
+	"fmt"
+
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/cadence"
+	"github.com/onflow/flow-go/cmd/util/cmd/common"
+	"github.com/onflow/flow-go/fvm/systemcontracts"
+	"github.com/onflow/flow-go/model/bootstrap"
+	model "github.com/onflow/flow-go/model/bootstrap"
+	"github.com/onflow/flow-go/model/cluster"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/model/flow/filter"
+	"github.com/onflow/flow-go/state/protocol/inmem"
+)
+
+// GenerateRecoverEpochTxArgs generates the required transaction arguments for the `recoverEpoch` transaction.
+func GenerateRecoverEpochTxArgs(log zerolog.Logger,
+	internalNodePrivInfoDir string,
+	nodeConfigJson string,
+	collectionClusters int,
+	epochCounter uint64,
+	rootChainID flow.ChainID,
+	numViewsInStakingAuction uint64,
+	numViewsInEpoch uint64,
+	targetDuration uint64,
+	targetEndTime uint64,
```

Comment:
> My concern is that this might not take epoch extensions into account (?)

You're right that it doesn't. That also caused a problem in the BlockTimeController PR -- see [this comment](https://github.com/onflow/flow-go/pull/6102#pullrequestreview-2166989320). I suggest we change all the `protocol.Epoch` methods that can be affected by extensions similarly to how `FinalView` was updated to reflect the `FinalView` of the latest extension.

Reply:
Fixed with @durkmurder PR https://github.com/onflow/flow-go/pull/6203